gax/src/main/java/com/google/api/gax/rpc/ClientContext.java

```diff
@@ -169,14 +197,22 @@ public static ClientContext create(StubSettings settings) throws IOException {
     ExecutorProvider backgroundExecutorProvider = settings.getBackgroundExecutorProvider();
     final ScheduledExecutorService backgroundExecutor = backgroundExecutorProvider.getExecutor();
 
-    Credentials credentials = settings.getCredentialsProvider().getCredentials();
-
-    if (settings.getQuotaProjectId() != null) {
-      // If the quotaProjectId is set, wrap original credentials with correct quotaProjectId as
-      // QuotaProjectIdHidingCredentials.
-      // Ensure that a custom set quota project id takes priority over one detected by credentials.
-      // Avoid the backend receiving possibly conflict values of quotaProjectId
-      credentials = new QuotaProjectIdHidingCredentials(credentials);
+    Credentials credentials = null;
+    Map<String, String> headers =
+        getHeadersFromSettingsAndEnvironment(settings, new SystemEnvironmentProvider());
```

Comment:
```suggestion
        getHeadersFromSettingsAndEnvironment(settings, System::getEnv);
```
and remove `SystemEnvironmentProvider`.

Reply:
Still pending.

Reply:
done